File: 03/part1.mjs
import { readFileSync } from 'fs'

/** @type {Array<Array<string>>} array[y][x] */
const input = readFileSync('./input.txt', { encoding: 'utf-8' })
  .split('\n')
  .filter(row => row.length)
  .map(row => row.split(''))
const width = input[0].length

let x = 0, y = 0
let vx = 3, vy = 1
let encounters = 0


do {
  if (input[y][x % width] === '#') encounters++

  x += vx
  y += vy
} while (y < input.length)

console.log(encounters)

File: 2024/01/part1.py
import sys
import pathlib

DEFAULT_FILE = pathlib.Path(__file__).parent.resolve().joinpath("input.txt")


def parse():
    a = []
    b = []

    for line in open(sys.argv[1] if len(sys.argv) > 1 else DEFAULT_FILE, "r"):
        row = line.split()
        a.append(int(row[0]))
        b.append(int(row[1]))

    return [a, b]


def main():
    [a, b] = parse()
    sum = 0

    a.sort()
    b.sort()

    for idx in range(len(a)):
        sum += abs(a[idx] - b[idx])

    print(sum)


if __name__ == "__main__":
    main()
